libvmi/libvmi.h

```diff
@@ -2465,6 +2477,29 @@ status_t vmi_set_vcpuregs(
     registers_t *regs,
     unsigned long vcpu) NOEXCEPT;
 
+/**
+ * Allocates a new physical page.
+ *
+ * @param[in] vmi LibVMI Instance
+ * @param[in] gfn gfn of physical page to be allocated
+ */
+status_t vmi_alloc_gfn(
+    vmi_instance_t vmi,
+    uint64_t gfn) NOEXCEPT;
+
+
+/**
+ * Frees a with vmi_alloc_gfn allocated physical page.
+ * Note that the maximum physical address cached by libvmi
+ * will not be updated accordingly.
```

Comment:
Why not adjust it if the gfn being removed is exactly where the `max_physical_address >> page_shift` is?

Reply:
I think that decreasing the cached value in this specific case might even be more confusing than not decreasing it at all. In order to correctly adjust the value, we would either have to keep track of freed pages, or always retrieve a new one via the hypervisor API. If you care about having a correct value, I would suggest the latter approach. Personally, I didn't think it was important enough to be implemented.